### PR TITLE
Changes to work with recent versions of Sphinx

### PR DESCRIPTION
**Pull request by peastman**
With these changes, it works with Sphinx 4.0.2.

### CODE REVIEW

**Comment by peastman** [comment]
This PR requires the changes to the sphinx extensions in https://github.com/rmcgibbo/sphinxcontrib-autodoc_doxygen/pull/22 and https://github.com/rmcgibbo/sphinxcontrib-lunrsearch/pull/3.  @rmcgibbo has tagged new releases, but he is no longer maintaining the conda packages.  We need someone take over doing that.

@Yoshanuikabundi is this something you might do?  As part of your overhaul of docs building, you may also want to look into eliminating use of sphinx extensions that are no longer maintained.

**Comment by peastman** [comment]
@rmcgibbo can you give @Yoshanuikabundi whatever permissions he needs so he can create the new packages on conda-forge?

File: devtools/ci/gh-actions/conda-envs/docs.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ dependencies:
 - swig
 - doxygen 1.8.14
 - pip:
-  - sphinx==2.3.1
-  - sphinxcontrib-bibtex<2.0.0
+  - sphinx==4.0.2
+  - sphinxcontrib-bibtex
   - sphinxcontrib-lunrsearch
   - sphinxcontrib-autodoc_doxygen
```
Comment (Yoshanuikabundi) [inline devtools/ci/gh-actions/conda-envs/docs.yml:19]: CI presently installs both of the affected packages via PIP, so the new tags have already taken effect. This PR is successfully building the docs, unlike `master` which was broken by the new releases of `sphinxcontrib-autodoc_doxygen` and `sphinxcontrib-lunrsearch`. The status failure is in a subsequent check that looks for broken links, not in building the docs.

I think this PR needs to be merged ASAP and is not blocked by upstream releases.

**Comment by peastman** [comment]
Ok, thanks!

**Comment by Yoshanuikabundi** [comment]
I'll look into replacing the unmaintained dependencies soon!

**Pull request merged**